packages/cxl-ui/src/components/cxl-inline-comment-context-menu.js

```diff
@@ -0,0 +1,152 @@
+      moduleId: 'cxl-inline-comment-context-menu',
+    });
+
+    this.appendContextMenuScript();
+  }
+
+  _appendOverlayCss() {
+    const overlayCss = `
+    [part="overlay"] {
+      border-radius: var(--lumo-border-radius-l);
+    }
+    [part="content"] {
+      width: var(--lumo-space-wide-xl);
+      color: transparent;
+      background: none;
+      box-shadow: none;
+      overflow: hidden;
+    }
+
+    `;
+    const style = document.createElement('style');
+    style.textContent = overlayCss;
+    this.$.overlay.shadowRoot.appendChild(style);
+  }
+
+  static get is() {
+    return 'cxl-inline-comment-context-menu';
+  }
+
+  static get properties() {
+    return {
+      ...super.properties,
+      textareaPlaceholder: {
+        type: String,
+        value: 'Put comment here',
+        notify: true,
+        reflectToAttribute: true,
+      },
+      buttonText: {
+        type: String,
+        value: 'Comment',
+        notify: true,
+        reflectToAttribute: true,
+      },
+      buttonTextSavingComment: {
+        type: String,
+        value: 'Saving...',
+        notify: true,
+        reflectToAttribute: true,
+      },
+    };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  appendContextMenuScript() {
+    const contextMenu = this;
+    contextMenu.closeOn = 'blur';
+
+    contextMenu.renderer = (root) => {
+      const listBox = this._getListBoxElementOrClearIfExists(root);
+
+      listBox.innerHTML = this._getListBoxDefaultHTML(listBox);
+
+      const sendButton = root.querySelector('#send');
+      const cancelButton = root.querySelector('#cancel');
+      const textarea = root.querySelector('.area-container textarea');
+
+      cancelButton.onclick = () => {
+        listBox.innerHTML = '';
+      };
+
+      sendButton.onclick = () => {
+        sendButton.disabled = true;
+        sendButton.textContent = this.buttonTextSavingComment;
+        this._dispachCustomEvent(textarea, contextMenu);
+
+        setTimeout(() => {
+          listBox.innerHTML = this._getListBoxSentHTML();
+
+          setTimeout(() => {
+            listBox.innerHTML = '';
+          }, 3000);
+        }, 2000);
+      };
+    };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  _getListBoxElementOrClearIfExists(root) {
+    let listBox = root.firstElementChild;
+
+    // // Check if there is a list-box generated with the previous renderer call to update its content instead of recreation
+    if (listBox) {
+      listBox.innerHTML = '';
+    } else {
+      listBox = window.document.createElement('vaadin-list-box');
+      root.appendChild(listBox);
+    }
+    return listBox;
+  }
+
+  _getListBoxDefaultHTML() {
+    return `
+          <div class="list-box-container">
+            <div class="area-container">
+                <textarea placeholder="${this.textareaPlaceholder}"></textarea>
+            </div>
+            <div class="buttons">
+              <button id="send">${this.buttonText}</button>
```

Comment:
Why are we not using `<vaadin-button>` and avoid a bunch of unnecessary styling?

Reply:
Fewer vaadin dependencies in custom components the better?

Reply:
No, it's the opposite - fewer random custom ideas - the better.

For component choice, it's also important to look at what's already loaded in main cxl-ui bundle. It is sometimes prudent to avoid importing single-use components and bloat bundle-size. But anything already loaded, should be used to maximum.

Reply:
noted

Reply:
@lkraav 
I'm having an issue with replacing regular `button` (which worked OK) for `vaadin-button` which doesn't work:
https://www.loom.com/share/e900c4d9fbf640409115d309de4319c6

Reply:
never mind, error fixed in: https://github.com/conversionxl/aybolit/pull/86/commits/1432229e0063ce1a489cc6ef21f40ccc1015cbe5